python/ray/autoscaler/launch_and_verify_cluster.py

```diff
@@ -333,18 +386,19 @@ def run_ray_commands(
         no_config_cache: Whether to pass the --no-config-cache flag to the ray CLI
             commands.
     """
-
+    provider_type = config_yaml.get("provider", {}).get("type")
+    if provider_type == "azure":
+        azure_authenticate()
```

Comment:
Calling this every time launch_and_verify_cluster.py gets called makes it difficult to run the script from as someone who doesn't have access to the AWS bucket with the secrets in there.

Would it makes sense to move the `azure_authenticate()` logic into a different script and call that before the `launch_and_verify_cluster.py` script in `release_tests.yaml` ?

It looks like the script reaches out to AWS for other cloud providers tho so maybe it would be better to make that change as a seperate PR if you think it is a good idea?